src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs

```diff
@@ -18,19 +18,274 @@
+using Moq;
+using Newtonsoft.Json;
 using SonarLint.VisualStudio.Core.Notifications;
+using SonarLint.VisualStudio.Core.VsVersion;
+using SonarLint.VisualStudio.Integration;
 using SonarLint.VisualStudio.Integration.UnitTests;
 
 namespace SonarLint.VisualStudio.Core.UnitTests.Notifications
 {
     [TestClass]
     public class DisabledNotificationsStorageTests
     {
+        private string ExpectedDisabledNotificationsFilePath => GetExpectedDisabledNotificationsFilePath();
+
         [TestMethod]
         public void MefCtor_CheckIsExported()
         {
-            MefTestHelpers.CheckTypeCanBeImported<DisabledNotificationsStorage, IDisabledNotificationsStorage>(null, null);
+            MefTestHelpers.CheckTypeCanBeImported<DisabledNotificationsStorage, IDisabledNotificationsStorage>(null, new[]
+            {
+                MefTestHelpers.CreateExport<IVsVersionProvider>(Mock.Of<IVsVersionProvider>()),
+                MefTestHelpers.CreateExport<ILogger>(Mock.Of<ILogger>())
+            });
+        }
+
+        [DataRow("2", true)]
+        [DataRow("4", false)]
+        [TestMethod]
+        public void IsNotificationDisabled_ReturnsCorrect(string notificationId, bool expectedResult)
+        {
+            var disabledNotifications = CreateDisabledNotifications("1", "2", "3");
+            var file = CreateFileMock(disabledNotifications);
+
+            var testSubject = CreateTestSubject(file.Object);
+                        
+            var result = testSubject.IsNotificationDisabled(notificationId);
+
+            result.Should().Be(expectedResult);
+            file.Verify(f => f.ReadAllText(ExpectedDisabledNotificationsFilePath), Times.Once);
+        }
+
+        [TestMethod]
+        public void IsNotificationDisabled_FileDoesNotExists_ReturnsFalse()
+        {
+            var file = CreateFileMock(fileExists: false);
+
+            var testSubject = CreateTestSubject(file.Object);
+
+            var result = testSubject.IsNotificationDisabled("1");
+
+            result.Should().BeFalse();
+            file.Verify(f => f.Exists(ExpectedDisabledNotificationsFilePath), Times.Once);
+            file.Verify(f => f.ReadAllText(ExpectedDisabledNotificationsFilePath), Times.Never);
+        }
+
+        [TestMethod]
+        public void IsNotificationDisabled_ReadError_LogsError()
+        {
+            var file = CreateFileMock();
+            file.Setup(f => f.ReadAllText(It.IsAny<string>())).Throws(new Exception("this is a test"));
+            var logger = new TestLogger();
+
+            var testSubject = CreateTestSubject(file.Object, logger);
+
+            var result = testSubject.IsNotificationDisabled("1");
+
+            result.Should().BeFalse();
+            logger.AssertOutputStrings(2);
+            logger.AssertPartialOutputStringExists("this is a test");
+        }
+
+        [TestMethod]
+        public void DisableNotification_IsNotDisabled_IsDisabledNotCalled_Disables()
```

Comment:
The name of this test is very confusing.

DisableNotification_NotificationIsNotPreviouslyDisabled_NotificationIsNowDisabled